applications/RANSApplication/python_scripts/formulations/utilities.py

```diff
@@ -0,0 +1,244 @@
+        #       is not used in this method at the moment.
+        #       Following is one of the proposals to make PeriodicConditions to work with connectivity_preserve_modeller.
+        # connectivity_preserve_modeler.GenerateModelPart(
+        #     origin_modelpart, model_part, element_name, condition_name,
+        #     original_condition_name + str(domain_size) + "D" + str(domain_size)
+        #     + "N")
+
+        connectivity_preserve_modeler.GenerateModelPart(
+            origin_modelpart, model_part, element_name, condition_name)
+
+    Kratos.Logger.PrintInfo("RANSModelPartFactory",
+                            "Created " + destination_modelpart_name)
+    return model.GetModelPart(destination_modelpart_name)
+
+
+def CreateResidualBasedBlockBuilderAndSolver(linear_solver, is_periodic,
+                                             communicator):
+    if (is_periodic):
+        return periodic_block_builder_and_solver(linear_solver, communicator)
+    else:
+        return block_builder_and_solver(linear_solver, communicator)
+
+
+def CreateResidualBasedNewtonRaphsonStrategy(model_part, scheme,
+                                             convergence_criteria,
+                                             builder_and_solver,
+                                             max_iterations, compute_reactions,
+                                             reform_dofs_at_each_step,
+                                             move_mesh_flag):
+    return newton_raphson_strategy(model_part, scheme,
+                                   convergence_criteria, builder_and_solver,
+                                   max_iterations, compute_reactions,
+                                   reform_dofs_at_each_step, move_mesh_flag)
+
+
+def CreateLinearSolver(solver_settings):
+    return linear_solver_factory.ConstructSolver(solver_settings)
+
+
+def CreateResidualCriteria(variable_convergence_criteria_info_list):
+    return residual_criteria(variable_convergence_criteria_info_list)
+
+
+def CreateIncrementalUpdateScheme():
+    return incremental_update_static_scheme()
+
+
+def CalculateNormalsOnConditions(model_part):
+    domain_size = model_part.ProcessInfo[Kratos.DOMAIN_SIZE]
+    if (not RansVariableUtilities.IsAnalysisStepCompleted(
+            model_part, "CONDITION_NORMAL_CALCULATION")):
+
+        # this calculates normals on whole model part, and assigns
+        # NORMAL variable in NodalSolutionStepDataValue container.
+        # NORMAL on conditions is required by some formulations such as inlet condition for
+        # incompressible potential flow velocity formulation, and all boundaries for incompressible
+        # potential flow pressure formulation.
+        Kratos.NormalCalculationUtils().CalculateOnSimplex(
+            model_part.Conditions, domain_size)
+        RansVariableUtilities.AddAnalysisStep(model_part,
+                                              "CONDITION_NORMAL_CALCULATION")
```

Comment:
Not for the discussion at hand, but I think we need to consensus before it spreads more and more. Would be nice to have something standard.